src/components/lib/posix/posix.c

```diff
@@ -293,6 +297,149 @@ cos_clone(int (*func)(void *), void *stack, int flags, void *arg, pid_t *ptid, v
 	return thd->thdid;
 }
 
+#define FUTEX_WAIT		0
```

Comment:
Is it possible to move all the futex implementation to a separate file?

Reply:
It's difficult with the current way libposix is setup as an object library, maybe doable though?

Reply:
ah, I see. the build system auto-generates .a files, so we'd have to generate special rules for having everything in a separate .c file. 

Another easy and better way is perhaps to have a "futex.h" header within "posix/" with all the implementation in it and include "futex.h" in posix.c (with some warning if already included). 

Or, at the very least, we could have the "#define", "struct"s in a separate header.

Reply:
It's actually more complex than that, because `libposix` is an object library... It isn't an easy fix, so I'm just going to leave it for now -- perhaps a future commit could fix this.

Reply:
@gparmer Yes, but it is in a very private header. (We probably could access it with some hacky include path, but I didn't think it was worth it.)